server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/RepositoryCache.java

```diff
@@ -69,71 +70,32 @@ public RepositoryCache(String cacheSpec, MeterRegistry meterRegistry) {
             builder.weigher((Weigher<CacheableCall, Object>) CacheableCall::weigh);
         }
         cache = builder.recordStats()
-                       .buildAsync((key, executor) -> {
-                           logger.debug("Cache miss: {}", key);
-                           return key.execute();
-                       });
+                       .buildAsync();
 
         CaffeineCacheMetrics.monitor(meterRegistry, cache, "repository");
     }
 
     public <T> CompletableFuture<T> get(CacheableCall<T> call) {
         requireNonNull(call, "call");
-        @SuppressWarnings("unchecked")
-        final CompletableFuture<T> f = (CompletableFuture<T>) cache.get(call);
-        return f;
-    }
-
-    @Nullable
-    public <T> CompletableFuture<T> getIfPresent(CacheableCall<T> call) {
-        requireNonNull(call, "call");
-        @SuppressWarnings("unchecked")
-        final CompletableFuture<T> f = (CompletableFuture<T>) cache.getIfPresent(call);
-        return f;
-    }
-
-    public <T> void put(CacheableCall<T> call, T value) {
-        requireNonNull(call, "call");
-        requireNonNull(value, "value");
-        cache.put(call, CompletableFuture.completedFuture(value));
-    }
-
-    public <T> T load(CacheableCall<T> key, Supplier<T> supplier, boolean logIfMiss) {
-        CompletableFuture<T> existingFuture = getIfPresent(key);
-        if (existingFuture != null) {
-            final T existingValue = existingFuture.getNow(null);
-            if (existingValue != null) {
-                // Cached already.
-                return existingValue;
-            }
+        final CompletableFuture<T> future = new CompletableFuture<>();
```

Comment:
Question) Should we check if the cache was populated by a previous call first before creating a new future?

Reply:
Creating a future is cheaper than call `Cache.getIfPresent`.

Reply:
This approach is recommended by the team. (Sorry for not giving you enough information.)
https://github.com/ben-manes/caffeine/wiki/Faq#recursive-computations

Reply:
I would like to know what your reasoning is. Could you elaborate on that? Is `.getIfPresent` more expensive cost than `.putIfAbsent`?

Reply:
If you want to check if the cache was popolulated, you need to call `getIfPresent` first and you need to call `put` or `putIfAbsent` if it's not created.
But in this approach, we can just call `putIfAbsent` only once. Did I miss something?

Reply:
I understand your intention. There is a trade-off for that. If the cache hit ratio is high, most calls will be handled by `getIfPresent()`.

As the change is trivial, let's keep it as it. I just wondered the reason. :-)

Reply:
That is also true.
Creating a future is relatively cheap and the code is shorter so I prefer this way. 😉